src/controllers/controller.polarArea.js

```diff
@@ -211,12 +223,29 @@ module.exports = function(Chart) {
 			return count;
 		},
 
-		calculateCircumference: function(value) {
+		computeAngle: function(value, index) {
+			var me = this;
 			var count = this.getMeta().count;
-			if (count > 0 && !isNaN(value)) {
-				return (2 * Math.PI) / count;
+			if (count <= 0 || isNaN(value)) {
+				return 0;
 			}
-			return 0;
+
+			var defaultAngle = (2 * Math.PI) / count;
+			var angleOption = me.chart.options.angle || defaultAngle;
+			var resolve = helpers.options.resolve;
+			var dataset = me.getDataset();
```

Comment:
I would change the method signature to only pass `index` as argument and read the value here:

```javascript
_computeAngle: function(index) {
   //...

  var dataset = me.getDataset();
  var value = dataset.data[index];

  //...
}
```